src/utils/utils_log.c

```diff
@@ -82,17 +85,73 @@ void util_log(util_log_level_t level, const char *format, ...) {
 #endif
 
     char buffer[LOG_MAX];
-    va_list args;
-    va_start(args, format);
-    int ret = vsnprintf(buffer, sizeof(buffer), format, args);
-    const char *overflow = "";
-    if (ret >= (intptr_t)sizeof(buffer)) {
+    char *b_pos = buffer;
+    int b_size = sizeof(buffer);
+
+    int tmp = vsnprintf(buffer, sizeof(buffer), format, args);
+    ASSERT(tmp > 0);
+
+    b_pos += (int)tmp;
+    b_size -= (int)tmp;
+
+    const char *postfix = "";
+
+    if (perror) {
```

Comment:
nit: could you move this to a separate function?

Reply:
Technically we can, but imho it's not wroth. This code is quite integrated with rest of the function, so it is a bit of extra code to abstract it, especial that i do not expect that we would need as a function to use in different case